pkg/es/config/config_test.go

```diff
@@ -0,0 +1,740 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+
+package config
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+)
+
+var mockEsServerResponseWithVersion0 = []byte(`
+{
+	"Version": {
+		"Number": "0"
+	}
+}
+`)
+
+var mockEsServerResponseWithVersion1 = []byte(`
+{
+	"tagline": "OpenSearch",
+	"Version": {
+		"Number": "1"
+	}
+}
+`)
+
+var mockEsServerResponseWithVersion2 = []byte(`
+{
+	"tagline": "OpenSearch",
+	"Version": {
+		"Number": "2"
+	}
+}
+`)
+
+func TestNewClient(t *testing.T) {
+	const (
+		pwd1        = "password"
+		token       = "token"
+		certContent = `-----BEGIN CERTIFICATE-----
```

Comment:
we already  have various test certificates, please use those
```
$ find . -name '*pem'
./pkg/config/tlscfg/testdata/example-server-cert.pem
./pkg/config/tlscfg/testdata/example-CA-cert.pem
./pkg/config/tlscfg/testdata/example-server-key.pem
./pkg/config/tlscfg/testdata/example-client-key.pem
./pkg/config/tlscfg/testdata/wrong-CA-cert.pem
./pkg/config/tlscfg/testdata/example-client-cert.pem
```

Reply:
i resolved it now